From a5404234dbd26e170acb97789599fbcd705cf6cb Mon Sep 17 00:00:00 2001
From: Xaver Hugl <xaver.hugl@gmail.com>
Date: Fri, 8 Apr 2022 17:47:09 +0200
Subject: [PATCH] backends/drm: don't permanently disable VRR when the test
 commit fails

In 5.24, the same code path is used for testing direct scanout, so that
causes false negatives. Generally though, the user setting shouldn't be
touched, it's not really proper feedback for the driver or KWin having
problems.


(cherry picked from commit 2ef6215a517e4220f147f5327e7ecdad02762b64)
---
 src/backends/drm/drm_output.cpp | 1 -
 1 file changed, 1 deletion(-)

diff --git a/src/backends/drm/drm_output.cpp b/src/backends/drm/drm_output.cpp
index bd08631cf7..048803883b 100644
--- a/src/backends/drm/drm_output.cpp
+++ b/src/backends/drm/drm_output.cpp
@@ -337,7 +337,6 @@ bool DrmOutput::present(const QSharedPointer<DrmBuffer> &buffer, QRegion damaged
             m_pipeline->applyPendingChanges();
         } else {
             m_pipeline->revertPendingChanges();
-            setVrrPolicy(RenderLoop::VrrPolicy::Never);
         }
     }
     if (m_pipeline->present(buffer)) {
-- 
GitLab

